sed/calibrator/delay.py

```diff
@@ -25,22 +28,102 @@ class DelayCalibrator:
     def __init__(
         self,
         config: dict = None,
-    ):
+    ) -> None:
         """Initialization of the DelayCalibrator class passes the config.
 
         Args:
             config (dict, optional): Config dictionary. Defaults to None.
         """
-        if config is None:
-            config = {}
-
-        self._config = config
+        self._config: dict = config or {}
+        self.loader: str = self._config["core"]["loader"]
 
-        self.adc_column = self._config["dataframe"]["adc_column"]
-        self.delay_column = self._config["dataframe"]["delay_column"]
-        self.calibration: Dict[Any, Any] = {}
+        self.adc_column: str = self._config["dataframe"].get("adc_column", None)
+        self.delay_stage_column: str = self._config["dataframe"].get("delay_stage_column", None)
```

Comment:
I don't see any benefit of doing this. You don't to a check in the functions where you use these if the values are non-none, and the default-entries should always be there (unless a user purposefully overwrites the default config dict...)

Reply:
removed